include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -80,12 +112,17 @@ namespace sparrow
     private:
 
         std::variant<ArrowArray*, ArrowArray> m_array;
+        ArrowArray& m_array_ref;
         std::variant<ArrowSchema*, ArrowSchema> m_schema;
+        ArrowSchema& m_schema_ref;
```

Comment:
`ArrowArray` and `ArrowSchema` should be pointers, otherwise the assignment operators will copy objects instead of reassigning the reference.

Reply:
done